.github/workflows/deploy-frontend.yml

```diff
@@ -4,36 +4,92 @@ on:
   workflow_call:
     inputs:
       report_artifact_in_scope:
-        required: false
         type: boolean
+        default: false
+      release_published:
+        type: boolean
+        default: false
   workflow_dispatch:
-  release:
-    types: [published]
+  pull_request:
+    paths:
+      - "frontend/**"
+      - ".github/workflows/deploy-frontend.yml"
 
 jobs:
+  # only build if triggered manually or new release published or a PR
   build-frontend:
-    uses: ./.github/workflows/build-frontend.yml
+    runs-on: ubuntu-latest
+    if: inputs.report_artifact_in_scope == false || inputs.release_published == true || github.event_name == 'pull_request'
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version-file: frontend/package.json
+
+      - uses: pnpm/action-setup@v4
+        with:
+          package_json_file: frontend/package.json
+          run_install: false
+
+      - name: Get pnpm store directory
+        shell: bash
+        run: |
+          echo "STORE_PATH=$(pnpm store path --silent)" >> $GITHUB_ENV
+
+      - uses: actions/cache@v4
+        name: Setup pnpm cache
+        with:
+          path: ${{ env.STORE_PATH }}
+          key: ${{ runner.os }}-pnpm-store-${{ hashFiles('**/pnpm-lock.yaml') }}
+          restore-keys: |
+            ${{ runner.os }}-pnpm-store-
+
+      - name: Install dependencies
+        run: pnpm install --dir frontend --frozen-lockfile --strict-peer-dependencies
+
+      - name: Build the UI
+        run: pnpm run --dir frontend build --base=${{ github.repository_owner == 'bowtie-json-schema' && '/' || '/bowtie' }}
+
+      - uses: actions/upload-artifact@v4
+        with:
+          name: frontend-build
+          path: frontend/build
+
+  # only start collecting if the build-frontend didn't fail (got skipped or successful) and the trigger of the workflow wasn't a PR
   collect:
     runs-on: ubuntu-latest
+    if: ${{ !failure() && github.event_name != 'pull_request' }}
     needs: build-frontend
     steps:
       - name: Create _site directory
         run: mkdir _site
-      - name: Download frontend build
-        uses: actions/download-artifact@v4
+
+      # if called from the report workflow and it wasn't a new release event (it was either a cron job or a manual dispatch of the report workflow)
+      - name: Download frontend-build from last release
+        uses: dawidd6/action-download-artifact@v3
+        if: inputs.report_artifact_in_scope == true && inputs.release_published == false
         with:
+          workflow: ci.yml
+          event: release
           name: frontend-build
           path: _site
```

Comment:
Mostly concerned about this one! Does it seem like the correct way ?

Reply:
I am not sure why you used `ci.yml` workflow here. But if you have used the `ci.yml` here, then the `frontend-build` artifact must be available in that workflow else this workflow run will fail. And as far as I can see, no `frontend-build` artifact is present in the `ci.yml`

Reply:
Have you tested the workflow? Is it running properly?

Reply:
> I am not sure why you used ci.yml workflow here. But if you have used the ci.yml here, then the frontend-build artifact must be available in that workflow else this workflow run will fail. And as far as I can see, no frontend-build artifact is present in the ci.yml.

This was the reason for my question on Slack @AgniveshChaubey . As I mentioned there I had observed that whenever a new release happens only `ci.yml` runs and the chaining of workflow happens within it. So at that time when a new release happens we build the frontend just like [here](https://github.com/bowtie-json-schema/bowtie/pull/1182/files#r1605786645) and because of workflow call chaining starting from the parent `ci.yml` this `deploy-frontend.yml` also gets triggered anyways but under the context of `ci.yml`. So that's the reason above here I wrote `ci.yml` to find the last frontend-build release.

Reply:
> Have you tested the workflow? Is it running properly?

Regarding this I've some issues. For testing this on my fork I wanted some release run of `ci.yml` on my fork to test but since I don't have one by default unlike here in Bowtie repo where multiple release runs of ci.yml are present, can you help me figure out how can I trigger a release on my fork so that `ci.yml` runs ? cc @Julian